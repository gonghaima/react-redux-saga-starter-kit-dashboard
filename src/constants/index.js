import keyMirror from "fbjs/lib/keyMirror";

const ActionTypes = keyMirror({
  EXCEPTION: undefined,
  GET_PRODUCTS: undefined,
  PRODUCT_GET_SUCCESS: undefined,
  PRODUCT_GET_FAILURE: undefined,
  SELECTION_CHANGE: undefined,
  PAGE_CHANGE: undefined,
  GET_USERS: undefined,
  USER_GET_SUCCESS: undefined,
  USER_GET_FAILURE: undefined,
  FILTER_USERS: undefined
});

const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error"
};

export { ActionTypes, STATUS };
