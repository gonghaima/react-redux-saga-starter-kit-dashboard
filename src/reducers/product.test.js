import reducer from './product';
import { ActionTypes } from '../constants';

describe('Product reducer', () => {
  it('should return the initial state', () => {
    expect(reducer.product(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.GET_PRODUCTS}`, () => {
    expect(
      reducer.product(undefined, {
        type: ActionTypes.GET_PRODUCTS
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.PRODUCT_GET_SUCCESS}`, () => {
    expect(
      reducer.product(undefined, {
        type: ActionTypes.PRODUCT_GET_SUCCESS,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.PRODUCT_GET_FAILURE}`, () => {
    expect(
      reducer.product(undefined, {
        type: ActionTypes.PRODUCT_GET_FAILURE,
        payload: {}
      })
    ).toMatchSnapshot();
  });
});
