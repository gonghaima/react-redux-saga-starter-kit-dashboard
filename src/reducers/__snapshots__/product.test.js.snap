// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product reducer should handle GET_PRODUCTS 1`] = `
Object {
  "data": Array [],
  "message": "",
  "query": "",
  "status": "running",
}
`;

exports[`Product reducer should handle PRODUCT_GET_FAILURE 1`] = `
Object {
  "data": Array [],
  "message": "",
  "query": "",
  "status": "error",
}
`;

exports[`Product reducer should handle PRODUCT_GET_SUCCESS 1`] = `
Object {
  "data": Array [],
  "message": "",
  "query": "",
  "status": "ready",
}
`;

exports[`Product reducer should return the initial state 1`] = `
Object {
  "data": Array [],
  "message": "",
  "query": "",
  "status": "idle",
}
`;
